tsdb/head_append.go

```diff
@@ -458,6 +462,33 @@ func (a *headAppender) AppendHistogram(ref storage.SeriesRef, lset labels.Labels
 	return storage.SeriesRef(s.ref), nil
 }
 
+func validateHistogram(h *histogram.Histogram) error {
+	checkSpans := func(sign string, spans []histogram.Span, buckets []int64) error {
+		var spanBuckets uint32
+		for n, span := range spans {
+			if n > 0 && span.Offset < 0 {
```

Comment:
While negative offset can be valid in some cases (n=0 and you have +1 and -1 spans), but let's start with a stricter check to be never negative.
```suggestion
			if span.Offset < 0 {
```

Reply:
Yeah, I was wondering about that. I didn't originally have the `n > 0` check, but the first negative span of the histogram exposed by [histogram_experiments exposer](https://github.com/beorn7/histogram_experiments/tree/master/cmd/exposer) (which I use to test) has a offset of `-26` so I was unsure if that's allowed.

Reply:
Oh yes you are right. First offset can be negative.